lib/sinon/util/core/extend.js

```diff
@@ -85,7 +85,18 @@ module.exports = function extend(target /*, sources */) {
     var sources = slice(arguments, 1);
 
     return extendCommon(target, sources, function copyValue(dest, source, prop) {
-        dest[prop] = source[prop];
+        var destOwnPropertyDescriptor = Object.getOwnPropertyDescriptor(dest, prop);
+        var sourceOwnPropertyDescriptor = Object.getOwnPropertyDescriptor(source, prop);
+
+        // If the property is 'name' but is not [[writable]], skip it
+        if (prop !== "name" || destOwnPropertyDescriptor.writable) {
+            Object.defineProperty(dest, prop, {
+                configurable: sourceOwnPropertyDescriptor.configurable,
+                enumerable: sourceOwnPropertyDescriptor.enumerable,
+                writable: sourceOwnPropertyDescriptor.writable,
+                value: sourceOwnPropertyDescriptor.value
+            });
+        }
```

Comment:
The  comment explains the idea, the implementation uses inversed logic, making it difficult to follow.

Instead, I suggest dropping the comment and make the implementation say what the comment said:

```suggestion
        if (prop === "name" && !destOwnPropertyDescriptor.writable) {
        	return;
        }

        Object.defineProperty(dest, prop, {
            configurable: sourceOwnPropertyDescriptor.configurable,
            enumerable: sourceOwnPropertyDescriptor.enumerable,
            writable: sourceOwnPropertyDescriptor.writable,
            value: sourceOwnPropertyDescriptor.value
        });
```